mid_algorithm.py

```diff
@@ -0,0 +1,75 @@
+"""
+This module contains a partially sophisticated algorithm
+"""
+
+""""""""""""""""""""""""""""""""""""""""""" Imports """""""""""""""""""""""""""""""""""""""""""
+from utils import *
+from abstract_algorithm import LearningAlgorithm
+
+""""""""""""""""""""""""""""""""""""""""""" Classes """""""""""""""""""""""""""""""""""""""""""
+
+
+class MaxVarianceAlgorithm(LearningAlgorithm):
```

Comment:
I would inherite from SequenceAlgorithm in naive algorithm (and move SequenceAlgorithm to abstract_algorithm.py file), but it's up to you :)

Reply:
I think I can't I need a diffrent fit stage
what do you think ?

Reply:
Indeed, but you can override on fit as virtual function